app/models/user.rb

```diff
@@ -275,6 +277,26 @@ def current_immersive_context_and_content_at(path_item)
     end
   end
 
+  def formal_first_name
+    verified_first_name || first_name
+  end
+
+  def formal_last_name
+    verified_last_name || last_name
+  end
+
+  def formal_full_name
+    "#{formal_first_name} #{formal_last_name}"
+  end
+
+  def certificates_in_organization(organization = Organization.current)
+    certificates.where certificate_program: CertificateProgram.where(organization: organization)
```

Comment:
I was going to propose to use a join like:
`user.certificates.joins(:certificate_program).where(certificate_programs: {organization: organization })`

but it surprised me that the query you've come up with does a sub-select internally, so I think in this case with such small tables it's pretty much the same :smile: